### DISCUSSION TITLE
DNS Check Successful, can't enforce HTTPS in Github Pages

### DISCUSSION THREAD

**Discussion by hewitson-j** [community/community#44160] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I'm deploying a website through Github pages. I've done this before but not enough to consider myself professional.

When I go to Github pages in my repository, I add the web address of my custom domain, and for a few seconds it performs a DNS check. When it says successful, I try to click the checkbox but the box is still grayed out.

When I refresh the page it goes back to doing a DNS check and then after a couple of seconds it says DNS check was successful.

I've checked with my DNS registrar and I've created an "A" record with the IP addresses for my site on Github pages and a CNAME record as well.

I tried removing the custom domain and adding it again and that's when it started to go through the DNS check and then success cycle every time I loaded the page, but it still won't unblock the Enforce HTTPS box.

**Comment by Joeeeeca** [comment]
Anyone know an answer for this at all?

**Comment by ahll19** [comment]
I don't reckon you've found an answer have you? I have the same issue

**Comment by CDProjects** [comment]
I am having the same issue - everything checks out - legit CNAME record, legit 4 x A records and DNS check successful - still can't enforce HTTPS.

**Comment by tjex** [comment]
Also having the same issue.
set the A and AAA records for the apex domain, set a cname for www subdomain to point to <user>.github.io. Waited 24 hours. 
The DNS provider (inwx) as seemingly updated as the old ip addresses that were registered to the domain pointed to a message saying "no content here yet", but now the ip address also shows a "DNS address could not be found" error, just like when tyring to visit my github pages.

**Reply by tjex** [reply]
mxlookup also shows that tjex.net points to githubs provided domains for A records... I'm thinking it could be an issue with the certificate. Which also just stalls at step 1...

**Comment by tjex** [comment]
may be worth looking at this. Your DNS needs to be able to sign certificates from Lets Encrypt. I've read some other forums and others have solved this issue by entering a CAA via their DNS provider

https://docs.github.com/en/pages/getting-started-with-github-pages/securing-your-github-pages-site-with-https#troubleshooting-certificate-provisioning-certificate-not-yet-created-error

**Comment by ahll19** [comment]
My problem was solved. What ended up working for me was setting the domain as `mydomain.com` under the **Custom domain** section of the settings.
With regards to DNS records I have:

- a TXT record verifying I own the domain
- a CNAME record pointing to the github pages site (hostname: `www.mydomain.com`, value: `www.ahll19.github.io`
- An ALIAS record (hostname: `mydomain.com`, value: `ahll19.github.io`)
- A CAA record which points to letsencrypt. I don't remember how I set this up, but it might've been through github itself

It was a messy process but I got it working in the end.

**Comment by hameezG** [comment]
Remove any redirected urls. 
i had the same issue and fixed it by removing  the redirect domain record.
I was using NameCheap as my domain provider and they had a default redirect for domain such as http/ domain
deleting this fixed the problem.

**Reply by arthurBricq** [reply]
This 🔼 ! This solved my problem. Thanks.

**Reply by ericcwlin** [reply]
Thx, it works!

My case is NameCheap auto generate `example.com` -> `http://www.example.com?from=@`

**Reply by Sir-Leon1** [reply]
Thanks , this solved my problem. Removed the redirect domain record in my DNS records. (hostinger)

**Comment by adoghaditsday** [comment]
If youre on CLoudflare you can enforce the rules for your domain if you dont have any established, to force HTTP
![Screenshot 2025-02-27 013908](https://github.com/user-attachments/assets/a448dfae-e365-4de7-8d89-8c827754efb7)

**Comment by annapowellsmith** [comment]
You have to enter "https://yourdomain.com" in the 'Custom domain' field, i.e. specify https at the start of the domain.

**Reply by Danakut** [reply]
This solved the issue for me, thank you! (No idea why they don't drop a mention/example of it in the docs.)

**Reply by jamesk99** [reply]
This was spot on advice. Thank you @annapowellsmith

**Reply by Tormentous** [reply]
Thank you, this worked!?

**Reply by theexpensiveexperience** [reply]
thanks, worked

**Reply by KenanAegean** [reply]
Thanks!

**Reply by brozena** [reply]
Worked!
